contacts/contactsyaml/provider.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository at
+// https://github.com/hyperledger-labs/perun-node
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contactsyaml
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/hyperledger-labs/perun-node"
+)
+
+// Provider represents a contacts provider that provides access to contacts stored in a yaml file.
+//
+// It generates of all contacts in the yaml file during initialization.  Read, Write and Delete
+// operations act only on the cached list of contacts and do not update the yaml file.
+// Use UpdateFile method to write the latest cache to yaml file, whenever required.
+//
+// It also stores an instance of wallet backend that will be used or encoding/decoding peer.Address.
+// All peer addresses in an instance of Provider should use the same wallet backend.
+type Provider struct {
+	*providerCache
+
+	filepath string
```

Comment:
i suggest contactsPath or contactsFilePath instead of filepath

Reply:
Updated to `contactsFilePath` in c702b0c2dac0a056026d1f4634aa6e59f14c1b20.